ItemList/Source/ItemList.DependencyInjection/Adapters/UnityAdapter.cs

```diff
@@ -12,25 +12,31 @@ public UnityAdapter(UnityContainer container)
         {
             _container = container;
         }
-        public void RegisterRequestScoped<TInterface, TImplementation>()
-            where TImplementation : TInterface
-        {
-            _container.RegisterType<TInterface, TImplementation>(new HierarchicalLifetimeManager());
-        }
+        public IDependencyInjectionContainer RegisterRequestScoped<TInterface, TImplementation>()
+            where TImplementation : TInterface 
+            => RegisterInterfaceImplementation<TInterface, TImplementation>(new HierarchicalLifetimeManager());
 
-        public void RegisterRequestScoped<TType>(Func<TType> implementationFactory)
-        {
-            _container.RegisterType<TType>(new InjectionFactory(_ => implementationFactory()));
-        }
+        public IDependencyInjectionContainer RegisterRequestScoped<TType>(Func<TType> implementationFactory) 
+            => RegisterType<TType>(new HierarchicalLifetimeManager(), new InjectionFactory(_ => implementationFactory()));
+
+        public IDependencyInjectionContainer RegisterSingleton<TInterface, TImplementation>() 
+            where TImplementation : TInterface 
+            => RegisterInterfaceImplementation<TInterface, TImplementation>(new ContainerControlledLifetimeManager());
 
-        public void RegisterSingleton<TInterface, TImplementation>() where TImplementation : TInterface
+        public IDependencyInjectionContainer RegisterSingleton<TType>(Func<TType> implementationFactory) 
+            => RegisterType<TType>(new ContainerControlledLifetimeManager(), new InjectionFactory(_ => implementationFactory()));
```

Comment:
You actually call `new InjectionFactory(_ => implementationFactory()` in both `RegisterSingleton` and `RegisterRequestScoped` methods, don't you? This goes against the purpose of the `RegisterType` helper method – please, move this piece of code down to the helper method.